2-mandatory/3-shopping-list.js

```diff
@@ -20,6 +20,21 @@ let pantry = {
 
 function createShoppingList(recipe) {
   // write code here
+
+  let missingIngredients = [];
+
+  // 2 lovely array methods which I learned here
+
+  const allPantryContents = Object.values(pantry).flat();
+
+  for (let i = 0; i < recipe.ingredients.length; i++) {
+    if (!allPantryContents.includes(recipe.ingredients[i]))
+      missingIngredients.push(recipe.ingredients[i]);
+  }
```

Comment:
This works fine, but it would also be quite a good opportunity to use `.filter()` on the recipe ingredients, e.g.:

```js
const missingIngredients = recipe.ingredients.filter(
  (ingredient) => !allPantryContents.includes(ingredient)
);
```